quick start test fails

in release/version2.8 (not sure when this got broken) when testing the quickstart (datawaveWebTest):
```
Test UID: EventQuery400BadRequest/IntentionalError
Test Description: Creates an error for EventQuery by omitting all required query params

HTTP Response Status Code: 500
HTTP Response ContentType: application/xml;charset=UTF-8

Test Total Time: 0.058184
Test Status: 
[X] FAILED - Expected Code: '400' Actual: '500'
```
The stack trace in server.log:
```
Caused by: java.lang.NullPointerException: Query string cannot be null
        at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:209)
        at datawave.webservice.query.QueryParametersImpl.validate(QueryParametersImpl.java:98)
        at datawave.webservice.query.runner.QueryExecutorBean.validateQuery(QueryExecutorBean.java:42)
        at datawave.webservice.query.runner.QueryExecutorBean.createQuery(QueryExecutorBean.java:623)
```

Can't replicate on release/version2.8. Thinking that a prior quickstart build got deployed for you in this case.

In release/version2.8, server.log should have this for the stacktrace instead:
```
Caused by: java.lang.NullPointerException: QueryParameter 'query' cannot be null
at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:209)
at datawave.webservice.query.QueryParametersImpl.validate(QueryParametersImpl.java:125)
```
Message text (and apparently the status code) got changed a while back
```
git log -p --follow -- \
web-services/client/src/main/java/datawave/webservice/query/QueryParametersImpl.java \
| grep -E "^commit|Query string cannot be null"

commit 30c9cef846f07624fd0cae648fc453501ee8f403
commit 055b21f0d0a2097b8f749797f162f3a2489dcfb1
commit b9a1c8bc4177641078d0a709808f11484d07dc83
-        Preconditions.checkNotNull(this.query, "Query string cannot be null");
commit 7f51e2668227f0a8ece3ae12d96ebfac43eafacf
commit 6c2d4310579f9ee5434640b4ff65c78656a3e2c0
commit 37148c47d004b05be27670799b67c4db53230410
commit 2f028f3f3f69e423506c1b8873183fdecac09f5e
+        Preconditions.checkNotNull(this.query, "Query string cannot be null");
```
The quickstart **allInstall** operation will prefer your previously-downloaded / -built tarballs, rather than forcing a re-download of dependencies and rebuild of DW. Likewise, **allUninstall** by default will retain all your tarballs to save time on reinstall attempts. You can override that with **allUninstall --remove-binaries** to completely start over from scratch

Admittedly, the quickstart could do a better job here. I think both allInstall and allIUinstall should be improved, so that the user gets advanced notice on exactly which tarballs are about to be operated on. We could do that under this ticket, or create a new one

--- a/contrib/datawave-quickstart/bin/common.sh
+++ b/contrib/datawave-quickstart/bin/common.sh
@@ -134,6 +134,16 @@ function allStart() {
    done
 }
 
+function allDisplayBinaryInfo() {
+   # Starts all registered services
+   local services=(${DW_CLOUD_SERVICES})
+   for servicename in "${services[@]}" ; do
+      echo "========================================================="
+      echo "$(printGreen "Binary info for ${servicename}")"
+      ${servicename}DisplayBinaryInfo
+   done
+}
+
 function allStop() {
 
    [[ "${1}" == "--hard" ]] && local kill9=true
@@ -218,10 +228,53 @@ function allInstall() {
       allStatus
       return 1
    fi
+   allDisplayBinaryInfo ; echo
    local services=(${DW_CLOUD_SERVICES})
+   if ! checkBinaries ; then
+      askYesNo "Please review the reported inconsistencies in your current setup.
+Hint: Existing local binaries will take precedence over source binaries during installation.
+If that's not what you want, please use $(printGreen "[servicename|all]Uninstall --remove-binaries") to remove local binaries first
+Continue with installation?" || return 1
+   fi
    for servicename in "${services[@]}" ; do
       ${servicename}Install
    done
+}
+
+function checkBinaries()  {
+  local status=0
+  # Dump source and local binaries into distinct arrays
+  local sourceBinaries=()
+  local localBinaries=()
+  while IFS= read -r line; do
+    if [[ "${line}" =~ ^Source:.* ]] ; then
+       sourceBinaries+=("$(cut -d: -f2- <<<${line})")
+    else
+       localBinaries+=("$(cut -d: -f2- <<<${line})")
+    fi
+  done < <( allDisplayBinaryInfo | grep -E 'Source:|Local:' )
+
+  # Compare each tarball pair
+  for ((i=0; i<${#sourceBinaries[@]}; i++)); do
+     local sbin="$(basename "${sourceBinaries[$i]}")"
+     local lbin="$(basename "${localBinaries[$i]}")"
+     if [[ "${sbin}" != "${lbin}" ]] && [[ ! "${lbin}" =~ .*Not\ loaded.* ]] ; then
+        warn "Source and Local binaries are inconsistent: Source == ${sbin}, Local == ${lbin}"
+        status=1
+     fi
+  done
+  return ${status}
+}
+
+function getDataWaveVersion() {
+  local pomFile="${DW_DATAWAVE_SOURCE_DIR}/pom.xml"
+  if [[ -f "${pomFile}" ]]; then
+     local version="$(grep -m 1 '<version>' "${pomFile}" 2>/dev/null | sed 's|.*<version>\(.*\)</version>.*|\1|')"
+     [[ -z "${version}" ]] && return 1
+  else
+     return 1
+  fi
+  echo "${version}"
 }
 
 function allUninstall() {

--- a/contrib/datawave-quickstart/bin/services/accumulo/bootstrap.sh
+++ b/contrib/datawave-quickstart/bin/services/accumulo/bootstrap.sh
@@ -257,3 +257,20 @@ function accumuloPidList() {
       echo "${DW_ACCUMULO_PID_LIST} ${DW_ZOOKEEPER_PID_LIST}"
    fi
 }
+
+function accumuloDisplayBinaryInfo() {
+  echo "Source: ${DW_ACCUMULO_DIST_URI}"
+  local tarballName="$(basename "$DW_ACCUMULO_DIST_URI")"
+  if [[ -f "${DW_ACCUMULO_SERVICE_DIR}/${tarballName}" ]]; then
+     echo " Local: ${DW_ACCUMULO_SERVICE_DIR}/${tarballName}"
+  else
+     echo " Local: Not loaded"
+  fi
+  echo "Source: ${DW_ZOOKEEPER_DIST_URI}"
+  tarballName="$(basename "$DW_ZOOKEEPER_DIST_URI")"
+  if [[ -f "${DW_ACCUMULO_SERVICE_DIR}/${tarballName}" ]]; then
+     echo " Local: ${DW_ACCUMULO_SERVICE_DIR}/${tarballName}"
+  else
+     echo " Local: Not loaded"
+  fi
+}

--- a/contrib/datawave-quickstart/bin/services/datawave/bootstrap-ingest.sh
+++ b/contrib/datawave-quickstart/bin/services/datawave/bootstrap-ingest.sh
@@ -332,3 +332,18 @@ function datawaveIngestCreateTableSplits() {
    # Force splits file creation for use by any upcoming jobs, to ensure proper partitioning
    ${DW_DATAWAVE_INGEST_HOME}/bin/ingest/generate-splits-file.sh
 }
+
+function datawaveIngestDisplayBinaryInfo() {
+  echo "Source: DataWave Ingest Version $(getDataWaveVersion)//$(datawaveIngestTarballName)"
+  local installedTarball="$(ls -1 ${DW_DATAWAVE_SERVICE_DIR}/$(basename ${DW_DATAWAVE_INGEST_TARBALL}) 2>/dev/null)"
+  if [[ -n "${installedTarball}" ]]; then
+     echo " Local: ${installedTarball}"
+  else
+     echo " Local: Not loaded"
+  fi
+}
+
+function datawaveIngestTarballName() {
+   local dwVersion="$(getDataWaveVersion)"
+   echo "$( basename "${DW_DATAWAVE_INGEST_TARBALL/-\*-/-$dwVersion-}" )"
+}

--- a/contrib/datawave-quickstart/bin/services/datawave/bootstrap-web.sh
+++ b/contrib/datawave-quickstart/bin/services/datawave/bootstrap-web.sh
@@ -160,3 +160,25 @@ function datawaveWebStart() {
     done
     return 1
 }
+
+function datawaveWebDisplayBinaryInfo() {
+  echo "Source: DataWave Web Version $(getDataWaveVersion)//$(datawaveWebTarballName)"
+  local installedTarball="$(ls -1 ${DW_DATAWAVE_SERVICE_DIR}/$(basename ${DW_DATAWAVE_WEB_TARBALL}) 2>/dev/null)"
+  if [[ -n "${installedTarball}" ]]; then
+     echo " Local: ${installedTarball}"
+  else
+     echo " Local: Not loaded"
+  fi
+  echo "Source: ${DW_WILDFLY_DIST_URI}"
+  local tarballName="$(basename "$DW_WILDFLY_DIST_URI")"
+  if [[ -f "${DW_DATAWAVE_SERVICE_DIR}/${tarballName}" ]]; then
+     echo " Local: ${DW_DATAWAVE_SERVICE_DIR}/${tarballName}"
+  else
+     echo " Local: Not loaded"
+  fi
+}
+
+function datawaveWebTarballName() {
+   local dwVersion="$(getDataWaveVersion)"
+   echo "$( basename "${DW_DATAWAVE_WEB_TARBALL/-\*-/-$dwVersion-}" )"
+}

--- a/contrib/datawave-quickstart/bin/services/datawave/bootstrap.sh
+++ b/contrib/datawave-quickstart/bin/services/datawave/bootstrap.sh
@@ -399,3 +399,8 @@ function datawaveBuild() {
    rm -f "${DW_DATAWAVE_SERVICE_DIR}"/datawave*.tar.gz
    resetQuickstartEnvironment
 }
+
+function datawaveDisplayBinaryInfo() {
+   datawaveIngestDisplayBinaryInfo
+   datawaveWebDisplayBinaryInfo
+}

--- a/contrib/datawave-quickstart/bin/services/hadoop/bootstrap.sh
+++ b/contrib/datawave-quickstart/bin/services/hadoop/bootstrap.sh
@@ -203,3 +203,13 @@ function hadoopPidList() {
    hadoopIsRunning && echo "${DW_HADOOP_PID_LIST}"
 
 }
+
+function hadoopDisplayBinaryInfo() {
+  echo "Source: ${DW_HADOOP_DIST_URI}"
+  local tarballName="$(basename "$DW_HADOOP_DIST_URI")"
+  if [[ -f "${DW_HADOOP_SERVICE_DIR}/${tarballName}" ]]; then
+     echo " Local: ${DW_HADOOP_SERVICE_DIR}/${tarballName}"
+  else
+     echo " Local: Not loaded"
+  fi
+}

--- a/contrib/datawave-quickstart/bin/services/maven/bootstrap.sh
+++ b/contrib/datawave-quickstart/bin/services/maven/bootstrap.sh
@@ -104,6 +104,16 @@ function mavenPrintenv() {
    echo
 }
 
+function mavenDisplayBinaryInfo() {
+  echo "Source: ${DW_MAVEN_DIST_URI}"
+  local tarballName="$(basename "$DW_MAVEN_DIST_URI")"
+  if [[ -f "${DW_MAVEN_SERVICE_DIR}/${tarballName}" ]]; then
+     echo " Local: ${DW_MAVEN_SERVICE_DIR}/${tarballName}"
+  else
+     echo " Local: Not loaded"
+  fi
+}
+
 # Eager-loading here since Maven is required to build DataWave,
 # as opposed to lazy-loading like the other services...
 

